hwmgr-plugins/api/server/nar-callback/server.go

```diff
@@ -358,31 +347,39 @@ func (n *NodeAllocationRequestCallbackServer) ProvisioningRequestCallback(ctx co
 		}, nil
 	}
 
-	// Check if this is a duplicate callback with the same status
-	currentCallbackStatus := pr.Annotations[ctlrutils.CallbackStatusAnnotation]
-	if currentCallbackStatus == string(request.Body.Status) {
-		n.Logger.Debug("Filtering out duplicate callback with same status",
-			"provisioningRequestName", request.ProvisioningRequestName,
-			"nodeAllocationRequestId", request.Body.NodeAllocationRequestId,
-			"status", request.Body.Status,
-			"currentStatus", currentCallbackStatus)
+	// Dedupe on (status, NAR id) so InProgress is handled once per NAR and
+	// same-status updates for the same NAR don't retrigger reconciliation.
+	prevStatus := ""
+	prevNar := ""
+	if pr.Annotations != nil {
+		prevStatus = pr.Annotations[ctlrutils.CallbackStatusAnnotation]
+		prevNar = pr.Annotations[ctlrutils.CallbackNodeAllocationRequestIdAnnotation]
+	}
+	newStatus := string(request.Body.Status)
+	newNar := request.Body.NodeAllocationRequestId
 
+	if prevStatus == newStatus && prevNar == newNar {
```

Comment:
I wonder if we should have a log message indicating that the ProvisioningRequest has received a new NodeAllocationRequestID?